Allow for functional properties.

brettryan opened the pull request:
Properties may be observables which need to be applied to determine the value.

Adding a suggestion to stop removing sort on a third click as it's redundant.

brettryan committed:
Allow for functional properties.

Properties may be observables which need to be applied to determine the value.

Adding a suggestion to stop removing sort on a third click as it's redundant.

Wedvich commented:
@brettryan I don't mind the change to the sort toggle, do you want to include that too? In that case, would you mind updating the readme to describe that functionality as well? Then I'd be happy to merge it in, and thanks for contributing! :smile:

brettryan committed:
Always change sort direction instead of removing sort.

brettryan committed:
Always change sort direction instead of removing sort.

brettryan committed:
Merge branch 'master' of github.com:brettryan/knockout.tablesort

brettryan committed:
Favour Element.closest native method if available.

Polyfill: https://developer.mozilla.org/en-US/docs/Web/API/Element/closest
Native support except IE/Edge at this point in time.

brettryan commented:
Sort directions will no longer be removed. I've also added the MDN polyfill for Element.closest instead of closestParentOrSelf favouring a native method if available. This avoids recursion and repeated toLowerCase on the tag name being matched.
